bit_manipulation/highest_power_of_2.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @author : vcnovaes<https://github.com/vcnovaes>
+ * Find the highest power of 2 smaller or equal a given number n
+ *
+ * @param {number} n - The input number
+ * @returns {number} the highest power of 2 smaller or equal n
+ *
+ * @example
+ * const result = highestPowerOf2(16); // Returns 16
+ * const result2 = highestPowerOf2(5);  // Returns 4
+ */
+
+export const highestPowerOf2 = (n: number) => {
+  let result = 0;
+  for (let i = n; i >= 1; i--) {
```

Comment:
This loop seems somewhat sketchy, given that it only reduces `n` by one in each iteration. Linear runtime in `n` is not acceptable, this should be logarithmic in `n` (and thus linear in the number of bits).